docs/migrating/from-gatsby.md

```diff
@@ -0,0 +1,235 @@
+
+```js
+import fs from 'fs'
+import { join } from 'path'
+import matter from 'gray-matter'
+import { parseISO, format } from 'date-fns'
+
+const postsDirectory = join(process.cwd(), 'content', 'blog')
+
+export function getPostBySlug(slug) {
+  const realSlug = slug.replace(/\\.md$/, '')
+  const fullPath = join(postsDirectory, `${realSlug}.md`)
+  const fileContents = fs.readFileSync(fullPath, 'utf8')
+  const { data, content } = matter(fileContents)
+  const date = format(parseISO(data.date), 'MMMM dd, yyyy')
+
+  return { slug: realSlug, frontmatter: { ...data, date }, content }
+}
+
+export function getAllPosts() {
+  const slugs = fs.readdirSync(postsDirectory)
+  const posts = slugs.map((slug) => getPostBySlug(slug))
+
+  return posts
+}
+```
+
+## Site Configuration
+
+With Gatsby, your site's metadata (name, description, etc) is located inside `gatsby-config.js`. This is then exposed through the GraphQL API and consumed through a `pageQuery` or a static query inside a component.
+
+With Next.js, we recommend creating a config file similar to below. You can then import this file anywhere without having to use GraphQL to access your site's metadata.
+
+```js
+// src/config.js
+
+export default {
+  title: 'Starter Blog',
+  author: {
+    name: 'Lee Robinson',
+    summary: 'who loves Next.js.',
+  },
+  description: 'A starter blog converting Gatsby -> Next.',
+  social: {
+    twitter: 'leeerob',
+  },
+}
+```
+
+## Search Engine Optimization
+
+Most Gatsby examples use `react-helmet` to assist with adding `meta` tags for proper SEO. With Next.js, we recommend using [`next/head`](/docs/api-reference/next/head) to add `meta` tags to your `<head />` element. For example, here's part of an SEO component with Gatsby:
+
+```js
+// src/components/seo.js
+
+import { Helmet } from 'react-helmet'
+
+return (
+  <Helmet
+    title={title}
+    titleTemplate={siteTitle ? `%s | ${siteTitle}` : null}
+    meta={[
+      {
+        name: `description`,
+        content: metaDescription,
+      },
+      {
+        property: `og:title`,
+        content: title,
+      },
+      {
+        property: `og:description`,
+        content: metaDescription,
+      },
+      {
+        property: `og:type`,
+        content: `website`,
+      },
+      {
+        name: `twitter:card`,
+        content: `summary`,
+      },
+      {
+        name: `twitter:creator`,
+        content: twitter,
+      },
+      {
+        name: `twitter:title`,
+        content: title,
+      },
+      {
+        name: `twitter:description`,
+        content: metaDescription,
+      },
+    ]}
+  />
+)
+```
+
+And here's the same example using Next.js, including reading from a site config file.
+
+```js
+// src/components/seo.js
+
+import Head from 'next/head'
+import config from '../config'
+
+const SEO = ({ description, title }) => {
```

Comment:
Could we use `function` instead? We’ve been using `function` throughout the docs.

```suggestion
export default function SEO({ description, title }) {
```